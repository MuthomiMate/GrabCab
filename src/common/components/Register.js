import React, { Component } from "react";
import { View, Text, Image, StyleSheet, TextInput, KeyboardAvoidingView, Button} from "react-native";
import { Icon } from "react-native-elements";

class Register extends Component {
    constructor(props){
        super(props)
        this.state= {
            password: "",
            confirmPassword: "",
            showPassword: true
        }
    }

    render(){
        const {password, confirmPassword, showPassword } =this.state
        return (
            <View style={styles.container}>
                <Image source = {require('../../images/back1.jpg')} style={styles.image}/>
                <View style={styles.textView}>
                    { /* <Image source = {require('../../images/car.png')} style={{width: "50%"}}/>*/ }
                    <Text style={styles.text}>
                        <Text style={{color: "#C4A751"}}>GRAB</Text>
                        <Text style={{color: "#FFFFFF"}}>CAB</Text>
                    </Text>
                </View>
                <KeyboardAvoidingView style={styles.registerView}>
                    <View style={styles.text}>
                        <Text style={{color: "#FFFFFF"}}>Registration</Text>
                    </View>
                    <View style={styles.input}>
                        <View style={styles.icon}>
                            <Icon name="user" size={25} type="font-awesome" color="#FFFFFF"/>
                        </View>
                        <View style={styles.border}>
                            <TextInput
                                style={{ height: "100%", color:"#FFFFFF"}}
                                placeholder="First Name"
                                placeholderTextColor="#FFFFFF"
                            />
                        </View>
                    </View>
                    <View style={styles.input}>
                        <View style={styles.icon}>
                            <Icon name="user" size={25} type="font-awesome" color="#FFFFFF"  />
                        </View>
                        <View style={styles.border}>
                            <TextInput
                                style={{ height: "100%", color:"#FFFFFF"}}
                                placeholder="Last Name"
                                placeholderTextColor="#FFFFFF"
                            />
                        </View>
                    </View>
                    <View style={styles.input}>
                        <View style={styles.icon}>
                            <Icon name="envelope-o" size={25} type="font-awesome" color="#FFFFFF" />
                        </View>
                        <View style={styles.border}>
                            <TextInput
                                style={{ height: "100%", color:"#FFFFFF"}}
                                placeholder="Email"
                                keyboardType="email-address"
                                placeholderTextColor="#FFFFFF"
                            />
                        </View>
                    </View>
                    <View style={styles.input}>
                        <View style={styles.icon}>
                            <Icon name="mobile" size={25} type="font-awesome" color="#FFFFFF" />
                        </View>
                        <View style={styles.border}>
                            <TextInput
                                style={{ height: "100%", color:"#FFFFFF"}}
                                placeholder="Mobile"
                                keyboardType="phone-pad"
                                placeholderTextColor="#FFFFFF"
                            />
                        </View>
                    </View>
                    <View style={styles.input}>
                        <View style={styles.icon}>
                            <Icon name="lock" size={25} type="font-awesome" color="#FFFFFF" />
                        </View>
                        <View style={styles.border}>
                            <TextInput
                                style={{ height: "100%", color:"#FFFFFF"}}
                                value={password}
                                onChangeText={(text) => this.setState({password: text})}
                                placeholder="Password"
                                secureTextEntry={showPassword}
                                placeholderTextColor="#FFFFFF"
                            />
                        </View>
                    </View>
                    <View style={styles.input}>
                        <View style={styles.icon}>
                            <Icon name="lock" size={25} type="font-awesome" color="#FFFFFF" />
                        </View>
                        <View style={styles.border}>
                            <TextInput
                                style={{ height: "100%", color:"#FFFFFF"}}
                                placeholder="Confirm password"
                                onChangeText={(text) => this.setState({confirmPassword: text})}
                                value={confirmPassword}
                                secureTextEntry={showPassword}
                                placeholderTextColor="#FFFFFF"
                            />
                        </View>
                    </View>
                    <View style={styles.buttonView} >
                        <Button
                            title="Register Now"
                            onPress={() => this.props.navigation.navigate('HomeDrawer')}
                            color="#FFFFFF"
                        />
                    </View>
                </KeyboardAvoidingView>

            </View>
        )
    }
}
const styles = StyleSheet.create({
    container: {
        position: 'absolute',
        top: 0,
        left: 0,
        width: '100%',
        height: '100%',
      },
      image: {
        width: "100%",
        height: "100%",
        flex:1,
    },
    registerView: {
        position: "absolute",
        bottom: "5%",
        width: "70%",
        left: "15%",
        alignItems: "center",
        textAlign: "center"
    },
    border: {
        borderBottomColor: "white",
        borderBottomWidth: 1,
        width: "85%"
    },
    input: {
        flex: 1,
        flexDirection: "row",
        margin: 10,
        width:"100%",
    },
    icon: {
        width: "10%",
        marginRight: 10
    },
    buttonView: {
        width:"60%",
        flex: 1,
        height: 40,
        marginTop: "5%",
        textAlign: "center",
        backgroundColor: "#1F81D4",
        borderRadius: 10,
        justifyContent: "space-around"
    },
    text: {
        fontSize: 40,
        fontWeight: "bold",
        fontFamily: 'Cochin',
    },
    textView: {
        position: "absolute",
        top: "20%",
        width: "100%",
        alignItems: "center",
        textAlign: "center",
    }
});
export default Register;