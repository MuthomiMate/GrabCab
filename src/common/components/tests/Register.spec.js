import React from "react";
import { shallow } from "enzyme";
import Register from "../Register";

describe("<Register />", () => {
    
    it("renders without crashing", () => {
        const wrapper = shallow(<Register/>);
        expect(wrapper).toMatchSnapshot();
    })
})