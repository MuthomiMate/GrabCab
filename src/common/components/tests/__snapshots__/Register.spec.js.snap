// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> renders without crashing 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/images/back1.jpg",
      }
    }
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "position": "absolute",
        "textAlign": "center",
        "top": "20%",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Cochin",
          "fontSize": 40,
          "fontWeight": "bold",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#C4A751",
          }
        }
      >
        GRAB
      </Text>
      <Text
        style={
          Object {
            "color": "#FFFFFF",
          }
        }
      >
        CAB
      </Text>
    </Text>
  </View>
  <KeyboardAvoidingView
    enabled={true}
    keyboardVerticalOffset={0}
    style={
      Object {
        "alignItems": "center",
        "bottom": "5%",
        "left": "15%",
        "position": "absolute",
        "textAlign": "center",
        "width": "70%",
      }
    }
  >
    <View
      style={
        Object {
          "fontFamily": "Cochin",
          "fontSize": 40,
          "fontWeight": "bold",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
          }
        }
      >
        Registration
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "margin": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 10,
            "width": "10%",
          }
        }
      >
        <Themed.Icon
          color="#FFFFFF"
          name="user"
          size={25}
          type="font-awesome"
        />
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "white",
            "borderBottomWidth": 1,
            "width": "85%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          placeholder="First Name"
          placeholderTextColor="#FFFFFF"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#FFFFFF",
              "height": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "margin": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 10,
            "width": "10%",
          }
        }
      >
        <Themed.Icon
          color="#FFFFFF"
          name="user"
          size={25}
          type="font-awesome"
        />
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "white",
            "borderBottomWidth": 1,
            "width": "85%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          placeholder="Last Name"
          placeholderTextColor="#FFFFFF"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#FFFFFF",
              "height": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "margin": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 10,
            "width": "10%",
          }
        }
      >
        <Themed.Icon
          color="#FFFFFF"
          name="envelope-o"
          size={25}
          type="font-awesome"
        />
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "white",
            "borderBottomWidth": 1,
            "width": "85%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          keyboardType="email-address"
          placeholder="Email"
          placeholderTextColor="#FFFFFF"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#FFFFFF",
              "height": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "margin": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 10,
            "width": "10%",
          }
        }
      >
        <Themed.Icon
          color="#FFFFFF"
          name="mobile"
          size={25}
          type="font-awesome"
        />
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "white",
            "borderBottomWidth": 1,
            "width": "85%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          keyboardType="phone-pad"
          placeholder="Mobile"
          placeholderTextColor="#FFFFFF"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#FFFFFF",
              "height": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "margin": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 10,
            "width": "10%",
          }
        }
      >
        <Themed.Icon
          color="#FFFFFF"
          name="lock"
          size={25}
          type="font-awesome"
        />
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "white",
            "borderBottomWidth": 1,
            "width": "85%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="#FFFFFF"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "color": "#FFFFFF",
              "height": "100%",
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "margin": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 10,
            "width": "10%",
          }
        }
      >
        <Themed.Icon
          color="#FFFFFF"
          name="lock"
          size={25}
          type="font-awesome"
        />
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "white",
            "borderBottomWidth": 1,
            "width": "85%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Confirm password"
          placeholderTextColor="#FFFFFF"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "color": "#FFFFFF",
              "height": "100%",
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#1F81D4",
          "borderRadius": 10,
          "flex": 1,
          "height": 40,
          "justifyContent": "space-around",
          "marginTop": "5%",
          "textAlign": "center",
          "width": "60%",
        }
      }
    >
      <Button
        color="#FFFFFF"
        onPress={[Function]}
        title="Register Now"
      />
    </View>
  </KeyboardAvoidingView>
</View>
`;
